packages/core/src/icon/utils/icon.classnames.ts

```diff
@@ -118,7 +78,7 @@ export function getUpdateSizeStrategy(size: string) {
     return SizeUpdateStrategies.NilSizeValue;
   }
 
-  if (isString(size) && isIconTshirtSizeClassname(size)) {
+  if (isString(size) && isTshirtSize(size)) {
```

Comment:
`isTshirtSize()` already guards against non-strings and nil values. I think it is safe to remove `isString()` from this conditional...

Reply:
At the moment only TypeScript guards it on compile time. It will loop through the array of sizes when it is executed in JavaScript despite the parameter type. On the other hand, we expect the value to be one of the t-shirt sizes in the array so if the value is not found in the array it will return `false`. We are safe.

Reply:
Line 77 guards it only against `nil` and empty strings.